pkg/azurefile/nodeserver.go

```diff
@@ -362,6 +408,44 @@ func (d *Driver) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRe
 		klog.V(2).Infof("volume(%s) mount %s on %s succeeded", volumeID, source, cifsMountPath)
 	}
 
+	// If runtime OS is not windows and protocol is not nfs, save mountInfo.json
+	if runtime.GOOS != "windows" && protocol != nfs {
+		// Check if mountInfo.json is already present at the targetPath
+		isMountInfoPresent, err := directvolume.VolumeMountInfo(cifsMountPath)
+		if err != nil && !os.IsNotExist(err) {
+			return nil, status.Errorf(codes.Internal, "Could not save direct volume mount info %s: %v", cifsMountPath, err)
+		}
+		if isMountInfoPresent != nil { // If already present, skip the operation
+			klog.V(2).Infof("NodeStageVolume: mount info for volume %s is already present on %s", volumeID, targetPath)
+		} else { // if not mounted, save mountInfo.json
+			mountFsType := cifs
+			ipAddr, err := net.ResolveIPAddr("ip", server)
```

Comment:
This is needed as of now, because inside a kata-cc pod, the kata agent (using nix:mount rust crate) performs the CIFS mount and it fails with EHOSTUNREACH, even though a command line `mount -t cifs` succeeds on the same VM. 
Reference: https://docs.rs/nix/latest/src/nix/mount/linux.rs.html#104-133